provider/lib/main.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright 2018 The Flutter Architecture Sample Authors. All rights reserved.
+// Use of this source code is governed by the MIT license that can be found
+// in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:provider_sample/app.dart';
+import 'package:todos_repository_simple/todos_repository_simple.dart';
+
+void main() {
+  const todoRepo = TodosRepositoryFlutter(
```

Comment:
It'd probably make more sense to pass these using providers instead. It's not necessary to make a provider for `FileStorage`, but it definitely is for `TodosRepositoryFlutter`

So we'd have:

```dart
runApp(
  Provider<TodosRepository>( // downcast
    create: (_) => TodosRepositoryFlutter(
      fileStorage: FileStorage(
        'provider_todos',
        getApplicationDocumentsDirectory,
      ),
    child: ProviderApp(),
  ),
);
```

Reply:
Could you discuss a bit more why that's important here? I don't really need to pass this down to any Widgets other than the `ProviderApp` and don't need to tie it to the Widget lifecycle at all, so it felt like that would just require more work when passing the Repository directly works fine and makes the requirement more explicit?

Reply:
Ah, well, my wording was a bit confusing.

It's not _important_. But this is typically the kind of object that would be inside a provider/locator usually, so I figured for the sake of making an example, it might be logical to do so.